src/consent.js

```diff
@@ -95,6 +96,34 @@ export function getConsentMetadata(element, policyId = 'default') {
   );
 }
 
+/**
+ * @param {!Element} element
+ * @param {?string=} opt_policyId
+ * @return {!Promise<!JsonObject>}
+ *    - consentMetadata
+ *    - consentString
+ *    - consentPolicyState
+ */
+export function collectConsents(element, opt_policyId) {
+  return Services.consentPolicyServiceForDocOrNull(element).then((policy) => {
+    const gettersOrNull = dict({
+      'consentMetadata': policy && policy.getConsentMetadataInfo,
+      'consentString': policy && policy.getConsentStringInfo,
+      'consentPolicyState': policy && policy.whenPolicyResolved,
+    });
+    if (!policy) {
+      return gettersOrNull;
+    }
+    return /** @type {!JsonObject} */ (Promise.all(
+      Object.keys(gettersOrNull).map((key) =>
+        gettersOrNull[key]
+          .call(policy, opt_policyId || 'default')
+          .then((value) => ({[key]: value}))
+      )
+    ).then((objs) => Object.assign.apply({}, objs)));
+  });
+}
+
```

Comment:
I extracted what `amp-iframe` does and refactored it somewhat.

I feel like there's a better name for this function, but I'm not sure what to call the returned object other than `consents`. Suggestions welcome.

Reply:
IMO `collectConsents` makes me think about actually prompting some consent UI. I think something more specific like `getCollectedConsentData` is more clear.

Reply:
Renamed to `getConsentDataToForward`.